### ISSUE TITLE
Once restoration has started, require as little user interaction as possible, so the user can "walk away"

### ISSUE THREAD

**Issue by t-m-w** [stevesoltys/backup#758]
Right now, when restoring a backup, there are a few interactions that - from a user perspective - may be considered unnecessary, or that could be frontloaded.

1. **"Re-installing apps" could proceed automatically to "Restoring backup"**. When reinstalling apps has completed, requiring the user to tap "Next" means that the user has to watch the device closely in order to proceed, which feels particularly unnecessary if there were no errors. Some apps may fail to install, and currently, this is where we display an error and present the user with relevant information as we have it. However, from a user perspective, it could make more sense to tell them all of this at the very end of the process.
2. **Restoring files**. This is currently a step that happens when the whole process is over. If it were to be one of the first steps, instead, then combined with 1 above, the user could potentially walk away and have everything restored by the time they get back - possibly along with a list of errors, of course, if any.

**Comment by grote** [comment]
> 1. However, from a user perspective, it could make more sense to tell them all of this at the very end of the process.

No, because when apps fail to install, *this* is the time where they can manually install the apps, so their data will get restored in the next step.

We could however, auto-proceed when there were no errors.

> 2.  If it were to be one of the first steps, instead, then combined with 1 above, the user could potentially walk away and have everything restored by the time they get back

Would be a bit tricky. We'd need app selection and file selection to happen first and then provide some sort of unified progress/result page that combines all information that currently is separate.

I personally don't consider all that high-priority, because I for one would like to follow the restore closely, so I am aware of any issues and any apps that may not have restored as expected.

**Comment by t-m-w** [comment]
> No, because when apps fail to install, this is the time where they can manually install the apps, so their data will get restored in the next step.

Why does it need to happen here / why does the data need to be restored in the next step?

> I personally don't consider all that high-priority

I get that. Typically, a user can do nothing if something fails, though. We don't even display actionable errors. So it seems like they'd be most interested in trying to move on with their life.

**Comment by grote** [comment]
> Why does it need to happen here / why does the data need to be restored in the next step?

You need to install apps before you can restore data for them.

> Typically, a user can do nothing if something fails, though.

In the case of installing apps, you can actually do something.

> So it seems like they'd be most interested in trying to move on with their life.

Maybe we need user research on this. In any case, re-doing the restore process as outlined would be a big undertaking, so ideally it is worth doing.

**Comment by t-m-w** [comment]
> Maybe we need user research on this.

Agree, let's pause this and see what comes out of user research if/when we have some.

**Comment by mattfbacon** [comment]
As a user, this has also bothered me. I think it would be best to save the apps that failed to install to a list, along with their data, and then the user could come back at any point and take action, instead of just when the option is shown after the apps were installed. That would also resolve some other issues, like the general impersistence and fragility of this menu, where you lose the ability to install the apps and to restore their data at all if the app gets terminated or you mis-input somehow.
